File: javascript/QuantumCollapseException.js
class QuantumCollapseException extends Error {
    constructor(id) {
        // C# : base($"CRITICAL FAILURE...")
        super(`CRITICAL FAILURE: Object with ID '${id}' has collapsed and exploded!`);
        this.name = "QuantumCollapseException";
    }
}
module.exports = QuantumCollapseException;

File: csharp/QuantumCollapseException.cs
using System;

namespace NDP_Odev4
{
    // Custom exception thrown when stability hits zero
    public class QuantumCollapseException : Exception
    {
        public QuantumCollapseException(string id) 
            : base($"CRITICAL FAILURE: Object with ID '{id}' has collapsed and exploded!") { }
    }
}